sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -435,120 +319,6 @@ export class EventHubReceiver extends LinkEntity {
-      const wasCloseInitiated = rheaReceiver && rheaReceiver.isItselfClosed();
-      // Clears the token renewal timer. Closes the link and its session if they are open.
-      // Removes the link and its session if they are present in rhea's cache.
-      await this._closeLink(rheaReceiver);
-      // We should attempt to reopen only when the receiver(sdk) did not initiate the close
-      let shouldReopen = false;
-      if (receiverError && !wasCloseInitiated) {
-        // if there was an error and it is retryable, recreate the link
-        const translatedError = translate(receiverError);
-        if (translatedError.retryable) {
-          shouldReopen = true;
-          logger.verbose(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is retryable. This is a candidate for re-establishing " +
-              "the receiver link.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          logger.verbose(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the receiver link.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!wasCloseInitiated) {
-        // there wasn't an error, and the client didn't initialize the close; recreate the link
-        shouldReopen = true;
-        logger.verbose(
-          "[%s] close() method of Receiver '%s' with address '%s' was not called. " +
-            "There was no accompanying error as well. This is a candidate for re-establishing " +
-            "the receiver link.",
-          this._context.connectionId,
-          this.name,
-          this.address
-        );
-      } else {
-        const state: any = {
-          wasCloseInitiated: wasCloseInitiated,
-          receiverError: receiverError,
-          _receiver: this._receiver
-        };
-        logger.verbose(
-          "[%s] Something went wrong. State of Receiver '%s' with address '%s' is: %O",
-          this._context.connectionId,
-          this.name,
-          this.address,
-          state
-        );
-      }
-
-      if (!shouldReopen) {
-        return;
-      }
-
-      const receiverOptions: CreateReceiverOptions = {
-        onMessage: (context: EventContext) => this._onAmqpMessage(context),
-        onError: (context: EventContext) => this._onAmqpError(context),
-        onClose: (context: EventContext) => this._onAmqpClose(context),
-        onSessionClose: (context: EventContext) => this._onAmqpSessionClose(context),
-        onSessionError: (context: EventContext) => this._onAmqpSessionError(context),
-        newName: true // prevents service from sending an error stating that the link is still open
```

Comment:
I'll do some testing, but was newName only needed to prevent the race condition of trying to create a new link while tearing down an existing link with the same name? Do we need to watch out for this when initializing?

Reply:
Good catch, the comment on the newName sure means what you are saying, but I have no prior experience here :(

Reply:
After looking through the code, it seems like we should be able to set this to true when initializing a receiver. Part of what makes me think this is safe is because today if the connection is disconnected and reconnects, all receivers will be recreated with a new name anyway (a uuid string). We could probably even get rid of the option altogether and always use a unique name but I haven't checked everywhere that this method is called to reason about it.